src/ResourceManager/Consumption/Commands.Consumption/Cmdlets/Budget/SetAzureRmConsumptionBudget.cs

```diff
@@ -0,0 +1,276 @@
+        [Parameter(ParameterSetName = ParameterSetNames.NotificationItemParameterSet, Mandatory = false, HelpMessage = HelpMessages.EndDate)]
+        [ValidateNotNullOrEmpty]
+        public DateTime? EndDate;
+
+        [Parameter(ParameterSetName = ParameterSetNames.SubscriptionItemParameterSet, Mandatory = false, HelpMessage = HelpMessages.ResourceGroupName)]
+        [Parameter(ParameterSetName = ParameterSetNames.NotificationItemParameterSet, Mandatory = false, HelpMessage = HelpMessages.ResourceGroupName)]
+        [ValidateNotNullOrEmpty]
+        [ResourceGroupCompleter]
+        public string ResourceGroupName;
+
+        [Parameter(ParameterSetName = ParameterSetNames.SubscriptionItemParameterSet, Mandatory = false, HelpMessage = HelpMessages.MeterFilter)]
+        [Parameter(ParameterSetName = ParameterSetNames.NotificationItemParameterSet, Mandatory = false, HelpMessage = HelpMessages.MeterFilter)]
+        [ValidateNotNullOrEmpty]
+        public string[] MeterFilter;
+
+        [Parameter(ParameterSetName = ParameterSetNames.SubscriptionItemParameterSet, Mandatory = false, HelpMessage = HelpMessages.ResourceFilter)]
+        [Parameter(ParameterSetName = ParameterSetNames.NotificationItemParameterSet, Mandatory = false, HelpMessage = HelpMessages.ResourceFilter)]
+        [ValidateNotNullOrEmpty]
+        public string[] ResourceFilter;
+
+        [Parameter(ParameterSetName = ParameterSetNames.SubscriptionItemParameterSet, Mandatory = false, HelpMessage = HelpMessages.ResourceGroupFilter)]
+        [Parameter(ParameterSetName = ParameterSetNames.NotificationItemParameterSet, Mandatory = false, HelpMessage = HelpMessages.ResourceGroupFilter)]
+        [ValidateNotNullOrEmpty]
+        public string[] ResourceGroupFilter;
+
+        [Parameter(ParameterSetName = ParameterSetNames.SubscriptionItemParameterSet, Mandatory = false, HelpMessage = HelpMessages.NotificationKey)]
+        [Parameter(ParameterSetName = ParameterSetNames.NotificationItemParameterSet, Mandatory = true, HelpMessage = HelpMessages.NotificationKey)]
+        [ValidateNotNullOrEmpty]
+        public string NotificationKey;
+
+        [Parameter(ParameterSetName = ParameterSetNames.SubscriptionItemParameterSet, Mandatory = false, HelpMessage = HelpMessages.NotificationEnabled)]
+        [Parameter(ParameterSetName = ParameterSetNames.NotificationItemParameterSet, Mandatory = false, HelpMessage = HelpMessages.NotificationEnabled)]
+        public SwitchParameter NotificationEnabled;
+
+        [Parameter(ParameterSetName = ParameterSetNames.SubscriptionItemParameterSet, Mandatory = false, HelpMessage = HelpMessages.NotificationDisabled)]
+        [Parameter(ParameterSetName = ParameterSetNames.NotificationItemParameterSet, Mandatory = false, HelpMessage = HelpMessages.NotificationDisabled)]
+        public SwitchParameter NotificationDisabled;
```

Comment:
What happens if the user gives both Enabled and Disabled? This is not a pattern PowerShell cmdlets follow, please follow the original comment: Only have one of either "Enable" or "Disable", and if the user doesn't provide the parameter, reset to default.  If this cmdlet is really a "Patch", let me know as the advice is different.

Reply:
Disable was removed.